### ISSUE TITLE
vmctl: step-interval not considered for `/api/v1/label/__name__/values` query

### ISSUE THREAD

**Issue by belm0** [VictoriaMetrics/VictoriaMetrics#5369]
### Describe the bug

The `/api/v1/label/__name__/values` query can be heavy for a long time range.

Even though `vmctl native` has `--vm-native-step-interval` (default `month`), it starts by making a `__name__/values` query for the entire time range.

### To Reproduce

```
./vmctl-prod vm-native -s \
  --vm-native-src-addr=http://src-victoria-metrics-single-server:8428 \
  --vm-native-dst-addr=http://dst-victoria-metrics-single-server:8428 \
  --vm-native-filter-time-start='2023-08-11T00:00:00Z' \
  --vm-native-filter-time-end='2023-10-11T00:00:00Z'
```

deadline exceeded error on the server:
> `error in "/api/v1/label/__name__/values?match[]={__name__!=""}&start=2023-08-11T00:00:00Z&end=2023-10-11T00:00:00Z": cannot obtain values for label "__name__": error during label values search on time range for labelName="__name__": error when searching for metricIDs for tagFilter in slow path: deadline exceeded; tagFilter=__name__=~".+"`

I'd expect that each step interval range will have a separate query of `__name__` values, which will be lighter on the migration source.  (E.g. 08-11 to 09-01, 09-01 to 10-01, 10-01 to 10-11.)

### Version

```
$ ./vmctl-prod --version
vmctl version vmctl-20230726-125412-heads-vmctl-api-v1-export-0-ga7d2c728a
```

**Comment by belm0** [comment]
Regarding the deadline exceeded, for our database even a query of all names for 1 month will exceed the 30s deadline.

I don't really want to increase `search.maxQueryDuration` for all queries during our migration just so `vmctl` can make this names query.  It would be nice if there was a specific duration flag for `/api/v1/label/__name__/`.

**Comment by hagen1778** [comment]
@dmitryk-dk would you mind checking this?

**Comment by dmitryk-dk** [comment]
> @dmitryk-dk would you mind checking this?

Yes, sure I will take a look!

**Comment by dmitryk-dk** [comment]
Hi @belm0 ! I want to make an improvement where we will discover `values` at certain time intervals. But at the same time, the minimum time period will not be more than one week because the `values` usually do not change significantly over time periods, especially if the filter is based on metric names. To a greater extent, there is a change in the metrics labels. However, even over short periods of time, changes in metric names are usually not significant.
Therefore, I wanted to clarify how much this improvement will make the migration process more efficient or faster for you?

**Comment by belm0** [comment]
I tried to explain in the original post, but the problem is that making a `__name__/values` query over a long time range can easily hit a timeout or other limit of the server.  I was forced to work around this by calling `vmctl-prod` individually per month, which defeats the purpose of its step-interval function, which is intended to handle this for you.

**Comment by dmitryk-dk** [comment]
> I tried to explain in the original post, but the problem is that making a `__name__/values` query over a long time range can easily hit a timeout or other limit of the server. I was forced to work around this by calling `vmctl-prod` individually per month, which defeats the purpose of its step-interval function, which is intended to handle this for you.

Thank you, for your explanation! We will discuss this improvement and will make a decision.

**Comment by hagen1778** [comment]
https://github.com/VictoriaMetrics/VictoriaMetrics/pull/5569 was merged and will be likely included into the next release.

**Comment by valyala** [comment]
> I don't really want to increase search.maxQueryDuration for all queries during our migration just so vmctl can make this names query. It would be nice if there was a specific duration flag for /api/v1/label/__name__/.

Such a flag is available at VictoriaMetrics starting from [v1.100.0](https://docs.victoriametrics.com/changelog/) release - see the description of `-search.maxLabelsAPIDuration` command-line flag at [resource usage limits docs](https://docs.victoriametrics.com/#resource-usage-limits).

[`vmctl`](https://docs.victoriametrics.com/vmctl.html) passes the requested time range to `/api/v1/label/__name__/values` requests starting from [v1.100.0](https://docs.victoriametrics.com/changelog/).

Closing the issue as resolved.
